crates/utils/re_uri/src/lib.rs

```diff
@@ -4,47 +4,19 @@
 //! `rerun://`, which is an alias for `rerun+https://`. These schemes are then
 //! converted on the fly to either `http://` or `https://`.
 
-use re_protos::remote_store::v0::storage_node_client::StorageNodeClient;
-use std::net::Ipv4Addr;
+mod endpoints;
+mod error;
 
-#[derive(thiserror::Error, Debug)]
-pub enum ConnectionError {
-    /// Native connection error
-    #[cfg(not(target_arch = "wasm32"))]
-    #[error("Connection error: {0}")]
-    Tonic(#[from] tonic::transport::Error),
-
-    #[error(transparent)]
-    ParseError(#[from] url::ParseError),
-
-    #[error("server is expecting an unencrypted connection (try `rerun+http://` if you are sure)")]
-    UnencryptedServer,
-
-    #[error(transparent)]
-    InvalidScheme(#[from] InvalidScheme),
-
-    #[error("unexpected endpoint: {0}")]
-    UnexpectedEndpoint(String),
-
-    #[error("unexpected opaque origin: {0}")]
-    UnexpectedOpaqueOrigin(String),
-
-    #[error("unexpected base URL: {0}")]
-    UnexpectedBaseUrl(String),
-
-    #[error("URL {url:?} cannot be loaded as a recording")]
-    CannotLoadUrlAsRecording { url: String },
-}
-
-#[derive(thiserror::Error, Debug)]
-#[error("invalid or missing scheme (expected `rerun(+http|+https)://`)")]
-pub struct InvalidScheme;
+pub use self::{
+    endpoints::{catalog::CatalogEndpoint, recording::RecordingEndpoint},
+    error::Error,
+};
 
 /// The different schemes supported by Rerun.
 ///
 /// We support `rerun`, `rerun+http`, and `rerun+https`.
 #[derive(Debug, PartialEq, Eq, Copy, Clone, Hash)]
-pub enum Scheme {
+enum Scheme {
```

Comment:
I can make those changes myself but heads up: I need to build a form-like UI for an `Origin`, with a scheme popup, a host text field, and a port integer text field. This means I'll need:
- `pub` access to `Scheme`
- ability to build an `Origin` from it's parts

Reply:
That sounds reasonable!